src/FluentMigrator/IMigrationProcessor.cs

```diff
@@ -29,8 +29,8 @@ public interface IMigrationProcessor : IQuerySchema, IDisposable
         string ConnectionString { get; }
 
         void Execute(string template, params object[] args);
-        DataSet ReadTableData(string schemaName, string tableName);
-        DataSet Read(string template, params object[] args);
+        IDataReader ReadTableData(string schemaName, string tableName);
+        IDataReader Read(string template, params object[] args);
```

Comment:
Is breaking for strict semver. There's no references to such a policy anywhere I could find.  But I would say that OSS should default to strict semver.  This is a breaking change to a public interface, and interface stability is paramount.  It may not be in common use, but this also presents a breaking change with the separately-delivered FluentMigrator.Runner package, and version-disagreement breakage there should be well-advertised.

* is the broken API in common use externally?  -- would require polling to determine, but I suspect not.
* internally, does it present a breaking change to older versions of other, separately released packages that